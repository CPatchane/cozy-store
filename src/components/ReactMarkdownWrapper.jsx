import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
