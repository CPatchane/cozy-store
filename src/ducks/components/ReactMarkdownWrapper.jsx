import React from 'react'
import ReactMarkdown from 'react-markdown'
import Emoji from 'emoji-js'

const emojiParser = new Emoji()

export const reactMarkdownRendererOptions = {
  link: props => (
    <a href={props.href} target='_blank'>
      {props.children}
    </a>
  ),
  heading: props => {
    return React.createElement(
      `h${props.level}`,
      { className: `md-title md-title--h${props.level}` },
      props.children
    )
  }
}

export const ReactMarkdownWrapper = ({ source, parseEmoji, className }) => (
  <ReactMarkdown
    source={parseEmoji ? emojiParser.replace_colons(source) : source}
    renderers={reactMarkdownRendererOptions}
    className={className}
  />
)

export default ReactMarkdownWrapper
