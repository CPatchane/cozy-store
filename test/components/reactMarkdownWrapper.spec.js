'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ReactMarkdownWrapper, {
  reactMarkdownRendererOptions
} from 'ducks/components/ReactMarkdownWrapper'

Enzyme.configure({ adapter: new Adapter() })

describe('ReactMarkdownWrapper component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <ReactMarkdownWrapper source="Test [link]() __strong__" />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should handle correctly markdown with emojis', () => {
    const component = shallow(
      <ReactMarkdownWrapper
        source={'### Heading 3\n\nThis is a `test` :tada:'}
        parseEmoji
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should not render if no source provided', () => {
    const component = shallow(<ReactMarkdownWrapper />).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should have correct link renderer options', () => {
    const component = shallow(
      <reactMarkdownRendererOptions.link href="#">
        <div />
      </reactMarkdownRendererOptions.link>
    ).getElement()
    expect(component).toMatchSnapshot()
  })
  it('should have correct heading renderer options', () => {
    const component = shallow(
      <reactMarkdownRendererOptions.heading level="3">
        MyHeading
      </reactMarkdownRendererOptions.heading>
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
