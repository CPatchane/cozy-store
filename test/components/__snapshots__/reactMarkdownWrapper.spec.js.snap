// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMarkdownWrapper component should be rendered correctly 1`] = `
<ReactMarkdown
  containerTagName="div"
  parserOptions={Object {}}
  renderers={
    Object {
      "Link": [Function],
    }
  }
  source="Test [link]() __strong__"
/>
`;

exports[`ReactMarkdownWrapper component should have correct Link renderer options 1`] = `
<a
  href="#"
  target="_blank"
>
  <div />
</a>
`;
